frontend/webapp/utils/ldap/forms.py

```diff
@@ -0,0 +1,122 @@
+"""Provides Flask-Security login forms for usage with LDAP auth backend."""
+from flask import current_app
+from flask import request
+from flask_security.confirmable import requires_confirmation
+from flask_security.forms import Form
+from flask_security.forms import get_form_field_label
+from flask_security.forms import NextFormMixin
+from flask_security.utils import config_value
+from flask_security.utils import encrypt_password
+from flask_security.utils import get_message
+from flask_security.utils import verify_and_update_password
+from ldap3.core.exceptions import LDAPExceptionError
+from ldap3.core.exceptions import LDAPSocketOpenError
+from werkzeug.local import LocalProxy
+from wtforms import BooleanField
+from wtforms import PasswordField
+from wtforms import StringField
+from wtforms import SubmitField
+
+_datastore = LocalProxy(lambda: current_app.extensions["security"].datastore)
+
+
+class LDAPLoginForm(Form, NextFormMixin):
+    """Login form for LDAP users."""
+
+    email = StringField("User ID")
+    password = PasswordField(get_form_field_label("password"))
+    remember = BooleanField(get_form_field_label("remember_me"))
+    submit = SubmitField(get_form_field_label("login"))
+
+    def __init__(self, *args, **kwargs):
+        """Init new LDAP login form."""
+        super(LDAPLoginForm, self).__init__(*args, **kwargs)
+        self._args = args
+        self._kwargs = kwargs
+        if not self.next.data:
+            self.next.data = request.args.get("next", "")
+        self.remember.default = config_value("DEFAULT_REMEMBER_ME")
+
+    def validate(self):
+        """Validate LDAP logins against AD."""
+        if not super(LDAPLoginForm, self).validate():
+            return False
+
+        if self.email.data.strip() == "":
+            self.email.errors.append(get_message("USERID_NOT_PROVIDED")[0])
+            return False
+
+        if self.password.data.strip() == "":
+            self.password.errors.append(get_message("PASSWORD_NOT_PROVIDED")[0])
+            return False
+
+        try:
+            admin_user = _datastore.get_user(1)
+            if self.email.data == admin_user.username:
```

Comment:
have you tested this case? Here you query the email of the user with id 1, which is perfectly fine (should be unique).

Reply:
however, in the local datastore, the username ("admin") is not the same as the email (which can be whatever)

Reply:
doesn't matter and its tested.. for users it will be email and admin will be whatever is put on the configuration

Reply:
ok.